xla/backends/gpu/runtime/command_buffer_cmd.h

```diff
@@ -277,6 +279,13 @@ class CommandBufferCmd {
   // updates, thus they need to be consistent across calls to the function.
   virtual BufferUseVector buffers() const = 0;
 
+  absl::Status add_resource_use(ResourceUse use) {
```

Comment:
AddResourceUse as it's not a simple getter/setter

Reply:
I have a better idea, instead of mutating commands you can build thunk->resources mapping ahead of time from thunk sequence, and then pass them in constructor. Prefer immutable types when possible.

Reply:
Fixed.